internal/services/v1/experimental.go

```diff
@@ -520,11 +535,32 @@ func (es *experimentalServer) BulkCheckPermission(ctx context.Context, req *v1.B
 		})
 	}
 
+	// Run the checks in parallel.
 	if err := tr.StartAndWait(); err != nil {
 		return nil, es.rewriteError(ctx, err)
 	}
 
-	return resp, nil
+	// Resort the results based on the incoming request order to make it easier to consume. This is not strictly
+	// necessary as each result contains the request that made it, but for the non-streaming version
+	// of this API, it makes sense.
+	sortedPairs := make([]*v1.BulkCheckPermissionPair, len(req.Items))
```

Comment:
super nit but perhaps we could order the `pairs` slice in-place so we don't allocate another one?

Reply:
Done